lib/dockerregistry/inventory.go

```diff
@@ -872,6 +880,63 @@ func getRegistryTagsFrom(req stream.ExternalRequest) (*google.Tags, error) {
 	return tags, nil
 }
 
+func getGManifestListWrapper(
+	req stream.ExternalRequest) (*GManifestList, error) {
+
+	var gManifestList *GManifestList
+
+	var getGManifestListCondition wait.ConditionFunc = func() (bool, error) {
+		var err error
+
+		gManifestList, err = getGManifestListFrom(req)
+
+		// We never return an error (err) in the second part of our return
+		// argument, because we don't want to prematurely stop the
+		// ExponentialBackoff() loop; we want it to continue looping until
+		// either we get a well-formed value, or until it hits
+		// ErrWaitTimeout. This is how ExponentialBackoff() uses the
+		// ConditionFunc type.
+		if err == nil &&
+			gManifestList != nil &&
+			len(gManifestList.Manifests) > 0 {
+
+			return true, nil
+		}
+		return false, nil
```

Comment:
If you get an error, I think you should log it, even if only at `klog.V(2)`

Reply:
Hmm I do log errors encountered by the child function `getGManifestListFrom()`... Still, I could log oddities such as not having encountered an error but finding that for example the child manifests list is empty. Ack